app/src/main/java/org/jellyfin/androidtv/details/EpisodeDetailsFragment.kt

```diff
@@ -24,32 +29,53 @@ class EpisodeDetailsFragment(private val data: Episode) : BaseDetailsFragment()
 
 	private suspend fun buildDetails() {
 
-		Log.i("EpisodeDetailsFragment", data.name)
+		Log.i("EpisodeDetailsFragment", episode.name)
 
-		val primaryImageUrl = TvApp.getApplication().apiClient.GetImageUrl(data.id, ImageOptions())
+		val primaryImageUrl = TvApp.getApplication().apiClient.GetImageUrl(episode.id, ImageOptions())
 		val primaryImageBitmap = getImageFromURL(primaryImageUrl)
 
 		val selector = ClassPresenterSelector().apply {
 			// Attach your media item details presenter to the row presenter:
-			val detailsDescriptionPresenter = DetailsDescriptionPresenter(title = data.name, subtitle = "TODO", body = data.description)
+			val detailsDescriptionPresenter = DetailsDescriptionPresenter(title = episode.name, subtitle = "TODO", body = episode.description)
+
+			val overviewRowPresenter = FullWidthDetailsOverviewRowPresenter(
+				detailsDescriptionPresenter,
+				primaryImageBitmap?.let { DetailsOverviewLogoPresenter() })
 
+			overviewRowPresenter.onActionClickedListener = OnActionClickedListener {
+				if (it is BaseAction) {
+					val action = it as BaseAction
+					action.onClick()
+				} else {
+					Log.e(LOG_TAG, "The clicked Action did not derive from BaseAction, this is unsupported!")
+				}
+			}
 
-			addClassPresenter(DetailsOverviewRow::class.java,
-				FullWidthDetailsOverviewRowPresenter(
-					detailsDescriptionPresenter,
-					primaryImageBitmap?.let { DetailsOverviewLogoPresenter() }))
+			addClassPresenter(DetailsOverviewRow::class.java, overviewRowPresenter)
 
 
 			addClassPresenter(ListRow::class.java, ListRowPresenter())
 		}
 		rowsAdapter = ArrayObjectAdapter(selector)
 
-		val detailsOverview = DetailsOverviewRow("Media Item Details").apply {
-			imageDrawable = BitmapDrawable(resources, primaryImageBitmap)
-			// Add images and action buttons to the details view
-			addAction(Action(1, "Continue from 12:10"))
-			addAction(Action(2, "Play from Beginning"))
+		val actionsAdapter = SparseArrayObjectAdapter()
+		actionsAdapter.apply {
```

Comment:
.apply could be added directly after creating the instance.

`val actionsAdapter = ArrayObjectAdapter().apply{ /*code */ }`

Reply:
Okay, will change it and remove the SparseArrayObjectAdapter